compiler/rustc_codegen_cranelift/src/constant.rs

```diff
@@ -242,9 +243,9 @@ pub(crate) fn codegen_const_value<'tcx>(
 
 fn pointer_for_allocation<'tcx>(
     fx: &mut FunctionCx<'_, '_, 'tcx>,
-    alloc: ConstAllocation<'tcx>,
+    alloc_id: AllocId,
 ) -> crate::pointer::Pointer {
-    let alloc_id = fx.tcx.create_memory_alloc(alloc);
+    let alloc = fx.tcx.global_alloc(alloc_id).unwrap_memory();
```

Comment:
Maybe move the `.inner().mutability` here and rename the variable as appropriate? It is only the mutability we care about here, not the allocation itself.

Reply:
That makes a very long chain though, I'd consider the code as-is cleaner. But I can change it if you prefer.